zhinst/toolkit/control/connection.py

```diff
@@ -97,6 +97,12 @@ def connect_device(self, serial=None, interface=None):
             ToolkitConnectionError if the could not be established. 
 
         """
+        if serial is None or not serial.lower().startswith("dev"):
+            raise ToolkitConnectionError(
+                f"Serial '{serial}' is invalid. It needs to have the form 'dev1234'."
+            )
+        serial = serial.lower()
```

Comment:
wouldn't it be better if we do a `serial.lower()` first and check the sanity of serial afterwards? i.e. whether if it starts dev or not.

Reply:
`serial` can also be `None` (there is at least such a unit test), so we anyway have to add some complication here...